messaging/WFLY-10518_artemis_no_local_broker.adoc

```diff
@@ -0,0 +1,168 @@
+= [WFLY-10518] Connections to remote Artemis broker must not require local Artemis instance
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          messaging,jms,openshift
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Connections to remote Artemis-based broker (including A-MQ 7) are handled in
+WildFly by the
+https://wildscribe.github.io/WildFly/13.0/subsystem/messaging-activemq/server/pooled-connection-factory/index.html[pooled-connection-factory]
+resource that is in the `messaging-activemq`.
+
+This `pooled-connection-factory` resource can be used to connect to the local
+ Artemis server that "owns" it or to a remote Artemis broker.
+
+However, this `pooled-connection-factory` resource always require the presence
+ of a local Artemis server even when it connects only to a remote Artemis broker.
+This use case will become more prevalent as WildFly is deployed on OpenShift to
+connect to a remote A-MQ 7 broker.
+
+The presence of a local Artemis server has impact on the memory and CPU
+ footprint of WildFly image on OpenShift.
+This RFE will improve WildFly so that a connection to a remote Artemis-based
+ broker will no longer require the presence of a local instance of Artemis.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10518[WFLY-10518]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1014[EAP7-1014]
+* https://issues.jboss.org/browse/EAP7-1006[EAP7-1006] is a RFE related to Java EE7 JMS resource
+definitions that will benefit from this RFE.
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
+
+=== Affected Projects or Components
+
+* This is a internal restructuring of the `messaging-activemq` subsystem that requires no changes
+from Artemis or other messaging components.
+
+=== Other Interested Projects
+
+== Requirements
```

Comment:
This feature must be transparent for end deployment, only server configuration changes are acceptable.

Will be this transparent standalone client too ? Like it will be able to do all the stuff it was able to do with embedded Artemis broker.

Reply:
For deployments and standalone clients that uses JMS by looking up resources from JNDI, that will be transparent.

Reply:
What about native api  and direct queue addressing ?

Reply:
Clients using ActiveMQ Core protocol are connecting directly to the server. this feature does not change anything about that.